src/hdmf/validate/validator.py

```diff
@@ -124,6 +124,8 @@ def get_type(data):
             if isinstance(data.dtype, list):
                 return [get_type(data[0][i]) for i in range(len(data.dtype))]
             if data.dtype.metadata is not None and data.dtype.metadata.get('vlen') is not None:
+                if data.shape == (0,) and data.dtype.metadata["vlen"] == str:
+                    return "utf"
```

Comment:
```suggestion
                # Empty array of strings
                if len(data) == 0 and data.dtype.metadata["vlen"] == str:
                    return "utf"
```

Reply:
I think checking for ``len`` instead of ``shape`` should also accommodate the case of an empty n-d array